atomic_reactor/util.py

```diff
@@ -626,3 +627,54 @@ def get_credentials(self, docker_registry):
         except KeyError:
             logger.warn('%s not found in .dockercfg', docker_registry)
             return {}
+
+
+def get_manifest_digests(image, insecure=False, dockercfg_path=None,
+                         versions=('v1', 'v2')):
+    """Return manifest digest for image.
+
+    :param image: ImageName, the remote image to inspect
+    :param insecure: bool, when True registry's cert is not verified
+    :param dockercfg_path: str, dirname of .dockercfg location
+    :param versions: tuple, which manifest schema versions to fetch digest
+
+    :return: dict, versions mapped to their digest
+    """
+    auth = None
+    if dockercfg_path:
+        dockercfg = Dockercfg(dockercfg_path).get_credentials(image.registry)
+
+        username = dockercfg.get('username')
+        password = dockercfg.get('password')
+        if username and password:
+            auth = requests.auth.HTTPBasicAuth(username, password)
+
+    context = '/'.join([x for x in [image.namespace, image.repo] if x])
+    tag = image.tag or 'latest'
+    url = 'https://{}/v2/{}/manifests/{}'.format(image.registry, context, tag)
```

Comment:
How about we enhance this method to also take the registry url from the plugin, that way we can implement the same logic: if a URI scheme is not defined, fallback to https, otherwise use whatever is already there. Thoughts?